plan/plans/perf.go

```diff
@@ -0,0 +1,1015 @@
+func NewPerfSchemaPlan(tableName string) (isp *PerfSchemaPlan, err error) {
+	switch strings.ToUpper(tableName) {
+	case tableSetupActors:
+	case tableSetupObjects:
+	case tableSetupInstrs:
+	case tableSetupConsumers:
+	case tableSetupTimers:
+	case tableStmtsCurrent:
+	case tableStmtsHistory:
+	case tableStmtsHistoryLong:
+	case tablePrepStmtsInstances:
+	case tableTransCurrent:
+	case tableTransHistory:
+	case tableTransHistoryLong:
+	case tableStagesCurrent:
+	case tableStagesHistory:
+	case tableStagesHistoryLong:
+	default:
+		return nil, errors.Errorf("table PERFORMANCE_SCHEMA.%s does not exist", tableName)
+	}
+	isp = &PerfSchemaPlan{
+		TableName: strings.ToUpper(tableName),
+	}
+	return
+}
+
+// CREATE TABLE if not exists performance_schema.setup_actors (
+// 		HOST			CHAR(60) NOT NULL  DEFAULT '%',
+// 		USER			CHAR(32) NOT NULL  DEFAULT '%',
+// 		ROLE			CHAR(16) NOT NULL  DEFAULT '%',
+// 		ENABLED			ENUM('YES','NO') NOT NULL  DEFAULT 'YES',
+// 		HISTORY			ENUM('YES','NO') NOT NULL  DEFAULT 'YES');
+func buildResultFieldsForSetupActors() (rfs []*field.ResultField) {
+	tbName := tableSetupActors
+	rfs = append(rfs, buildUsualResultField(tbName, "HOST", mysql.TypeString, 60, mysql.NotNullFlag, "%"))
+	rfs = append(rfs, buildUsualResultField(tbName, "USER", mysql.TypeString, 32, mysql.NotNullFlag, "%"))
+	rfs = append(rfs, buildUsualResultField(tbName, "ROLE", mysql.TypeString, 16, mysql.NotNullFlag, "%"))
+	rfs = append(rfs, buildEnumResultField(tbName, "ENABLED", []string{"YES", "NO"}, mysql.NotNullFlag, "YES"))
+	rfs = append(rfs, buildEnumResultField(tbName, "HISTORY", []string{"YES", "NO"}, mysql.NotNullFlag, "YES"))
+	return rfs
+}
+
+// CREATE TABLE if not exists performance_schema.setup_objects (
+// 		OBJECT_TYPE		ENUM('EVENT','FUNCTION','TABLE') NOT NULL  DEFAULT 'TABLE',
+// 		OBJECT_SCHEMA	VARCHAR(64)  DEFAULT '%',
+// 		OBJECT_NAME		VARCHAR(64) NOT NULL  DEFAULT '%',
+// 		ENABLED			ENUM('YES','NO') NOT NULL  DEFAULT 'YES',
+// 		TIMED			ENUM('YES','NO') NOT NULL  DEFAULT 'YES');
+func buildResultFieldsForSetupObjects() (rfs []*field.ResultField) {
+	tbName := tableSetupObjects
+	rfs = append(rfs, buildEnumResultField(tbName, "OBJECT_TYPE", []string{"EVENT", "FUNCTION", "TABLE"}, mysql.NotNullFlag, "TABLE"))
+	rfs = append(rfs, buildUsualResultField(tbName, "OBJECT_SCHEMA", mysql.TypeVarchar, 64, 0, "%"))
+	rfs = append(rfs, buildUsualResultField(tbName, "OBJECT_NAME", mysql.TypeVarchar, 64, mysql.NotNullFlag, "%"))
+	rfs = append(rfs, buildEnumResultField(tbName, "ENABLED", []string{"YES", "NO"}, mysql.NotNullFlag, "YES"))
+	rfs = append(rfs, buildEnumResultField(tbName, "TIMED", []string{"YES", "NO"}, mysql.NotNullFlag, "YES"))
+	return rfs
+}
+
+// CREATE TABLE if not exists performance_schema.setup_instruments (
+// 		NAME			VARCHAR(128) NOT NULL,
+// 		ENABLED			ENUM('YES','NO') NOT NULL,
+// 		TIMED			ENUM('YES','NO') NOT NULL);
+func buildResultFieldsForSetupInstrs() (rfs []*field.ResultField) {
```

Comment:
I considered it, but looks like `Instruments` is a little bit long, also I used `trans` instead of `transactions`, of coz I can change if you insist.